target/ppc/translate/fp-impl.c.inc

```diff
@@ -1473,26 +1473,50 @@ static void gen_stfqx(DisasContext *ctx)
     tcg_temp_free_i64(t1);
 }
 
-static bool trans_PLFS(DisasContext *ctx, arg_PLFS *a)
+static bool do_plstfs(DisasContext *ctx, arg_D *d, bool store)
 {
     TCGv ea;
     TCGv_i64 t0;
-    arg_D d;
-    if (!resolve_PLS_D(ctx, &d, a)) {
-        return true;
-    }
+
     REQUIRE_FPU(ctx);
     gen_set_access_type(ctx, ACCESS_FLOAT);
     ea = tcg_temp_new();
     t0 = tcg_temp_new_i64();
-    do_ea_calc(ctx, d.ra, tcg_constant_tl(d.si), ea);
-    gen_qemu_ld32fs(ctx, t0, ea);
-    set_fpr(d.rt, t0);
+    do_ea_calc(ctx, d->ra, tcg_constant_tl(d->si), ea);
+
+    if (store)
+    {
+        get_fpr(t0, d->rt);
+        gen_qemu_st32fs(ctx, t0, ea);
+    }else{
+        gen_qemu_ld32fs(ctx, t0, ea);
+        set_fpr(d->rt, t0);
+    }
+
     tcg_temp_free(ea);
     tcg_temp_free_i64(t0);
     return true;
 }
 
+static bool trans_PLFS(DisasContext *ctx, arg_PLFS *a)
+{
+    arg_D d;
+    if (!resolve_PLS_D(ctx, &d, a)) {
+        return true;
+    }
+
+    return do_plstfs(ctx, &d, false);
+}
+
+static bool trans_PSTFS(DisasContext *ctx, arg_PSTFS *a){
```

Comment:
Ops.. It looks like the open curly bracket is on the same line of function.

Reply:
Corrected in the new version